### PR TITLE
BuildResidentialHPXML measure: Generic method for getting argument values

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Make getting measure argument values generic so we don't have to explicitly call out every argument string. Longer term, a method like this should probably be available in the OpenStudio API.

## Checklist

Not all may apply:

- [ ] EPvalidator.xml has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI

### CODE REVIEW

File: HPXMLtoOpenStudio/resources/meta_measure.rb [hunk 1]
```diff
@@ -440,3 +440,38 @@ def is_integer?
...
+        args[argument.name] = runner.getOptionalDoubleArgumentValue(argument.name, user_arguments)
+      when 'Integer'.to_OSArgumentType
+        args[argument.name] = runner.getOptionalIntegerArgumentValue(argument.name, user_arguments)
+      when 'String'.to_OSArgumentType
+        args[argument.name] = runner.getOptionalStringArgumentValue(argument.name, user_arguments)
+      end
+    end
+  end
+  args = Hash[args.collect { |k, v| [k.to_sym, v] }]
+  return args
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/meta_measure.rb:476]: 👍

**Comment by joseph-robertson** [comment]
> Seems like this should be inside the `get_argument_values` method.

It works better outside of the method. For example, https://github.com/NREL/resstock/blob/restructure-v3-measure-args/measures/ResStockArguments/measure.rb#L88-L93.

File: BuildResidentialHPXML/measure.rb [hunk 2]
```diff
@@ -4725,51 +4243,23 @@ def self.set_hvac_distribution(hpxml, runner, args)
...
-      setpoint_heating_weekend_schedule = args[:setpoint_heating_weekend_schedule].get.split(',').map { |i| Float(i) }
-      weekend_heating_setpoints = modify_setpoint_schedule(weekend_heating_setpoints, setpoint_heating_weekend_offset_magnitude, setpoint_heating_weekend_schedule)
-    end
-
-    if args[:setpoint_cooling_weekday_offset_magnitude].is_initialized && args[:setpoint_cooling_weekday_schedule].is_initialized
-      setpoint_cooling_weekday_offset_magnitude = args[:setpoint_cooling_weekday_offset_magnitude].get
-      setpoint_cooling_weekday_schedule = args[:setpoint_cooling_weekday_schedule].get.split(',').map { |i| Float(i) }
-      weekday_cooling_setpoints = modify_setpoint_schedule(weekday_cooling_setpoints, setpoint_cooling_weekday_offset_magnitude, setpoint_cooling_weekday_schedule)
+    if args[:setpoint_heating_weekday] == args[:setpoint_heating_weekend] && !args[:setpoint_heating_weekday].include?(',')
+      heating_setpoint_temp = args[:setpoint_heating_weekday]
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:4250]: So if they enter a single value and it's the same for weekdays and weekends, it gets written to the standard HPXML `SetpointTempHeatingSeason` and `SetpointTempCoolingSeason` elements right? And if not, it uses the 24-hour extension elements? Big 👍 if so.

File: BuildResidentialHPXML/measure.rb [hunk 3]
```diff
@@ -3081,414 +3007,6 @@ def run(model, runner, user_arguments)
...
-             pool_heater_annual_kwh: runner.getStringArgumentValue('pool_heater_annual_kwh', user_arguments),
-             pool_heater_annual_therm: runner.getStringArgumentValue('pool_heater_annual_therm', user_arguments),
-             pool_heater_usage_multiplier: runner.getDoubleArgumentValue('pool_heater_usage_multiplier', user_arguments),
-             hot_tub_present: runner.getBoolArgumentValue('hot_tub_present', user_arguments),
-             hot_tub_pump_annual_kwh: runner.getStringArgumentValue('hot_tub_pump_annual_kwh', user_arguments),
-             hot_tub_pump_usage_multiplier: runner.getDoubleArgumentValue('hot_tub_pump_usage_multiplier', user_arguments),
-             hot_tub_heater_type: runner.getStringArgumentValue('hot_tub_heater_type', user_arguments),
-             hot_tub_heater_annual_kwh: runner.getStringArgumentValue('hot_tub_heater_annual_kwh', user_arguments),
-             hot_tub_heater_annual_therm: runner.getStringArgumentValue('hot_tub_heater_annual_therm', user_arguments),
-             hot_tub_heater_usage_multiplier: runner.getDoubleArgumentValue('hot_tub_heater_usage_multiplier', user_arguments) }
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:3489]: 400 LOC removed. Awesome.

**Review by shorowit — APPROVED** [review]
Looks great to me.

File: BuildResidentialHPXML/measure.rb (hunk 2 above)
Comment (joseph-robertson) [reply BuildResidentialHPXML/measure.rb:4250]: That's the intent, yes.

**Pull request merged**